### PR TITLE
Refs: 18F/foia-hub#84 Updating usagov script to only handle data pull only.

### PR DESCRIPTION
**Pull request by jackiekazil**
Related to: 18F/foia-hub#142 & 18F/foia-hub#84
- Updating usagov script to only handle data pull only.
- Updating data in the repo w/ a fresh pull.
- Created sample data set from the full set. Adding it to the repo.

### CODE REVIEW

File: usa-contacts/usagov.py [hunk 1]
```diff
@@ -25,89 +38,69 @@ def grab_and_save_data():
...
     '''
         Example: create_sample_file([0, 100, 250, 400], 'data/all_data.json')
         Pulls indexed records from full datasets to create sample data.
     '''
 
+    # Load file with all data
+    try:
+        data = json.load(open(data_source))
+    except FileNotFoundError:
+        grab_and_save_data()
```
Comment (cmc333333) [inline usa-contacts/usagov.py:55]: I think that, if an exception is thrown, `data` won't be instatiated, so the interpreter will explode a few lines down.

**Comment by cmc333333** [comment]
It looks like the defaults pulls five records; will it eventually default to "all", or is that not the goal?

**Comment by jackiekazil** [comment]
The default is for all records. It only creates a sample file if you specify it to. 
See: https://github.com/18F/foia/blob/usacontacts/usa-contacts/usagov.py#L71-L89

**Comment by jackiekazil** [comment]
@cmc333333 I will fix the error as discussed, and then assign to you when I am done.

**Comment by cmc333333** [comment]
Coolio. It looks good otherwise; my mistake on misreading the defaults

**Comment by jackiekazil** [comment]
@cmc333333, can you review and merge? ... for now, I am done with this folder, so work on 18F/foia-hub#143 won't disrupt what I am doing.

File: usa-contacts/usagov.py [hunk 2]
```diff
@@ -15,6 +17,17 @@ def _get_data_folder():
...
 
 
+def setup_data_dir():
+    '''
+        Setup a data dir if it doesn't exist.
+    '''
+    # Make a data directory if it doesn't exist
+    if not os.path.isdir("data"):
+        os.mkdir("data")
+        return True
```
Comment (khandelwal) [inline usa-contacts/usagov.py]: Why does this return True/False values?

File: usa-contacts/usagov.py [hunk 3]
```diff
@@ -25,89 +38,70 @@ def grab_and_save_data():
...
+        data = json.load(open(data_source))
+    except FileNotFoundError:
+        grab_and_save_data()
+        data = json.load(open(data_source))
+
     samples = []
-    #TODO: Add data file load.
-    for n in sample_numbers:
+
+    for n in sample_recs:
```
Comment (khandelwal) [inline usa-contacts/usagov.py:60]: You  can replace these lines with 

`samples = [data[n] for n in sample_recs]`

for brevity if you like.

**Comment by khandelwal** [comment]
This doesn't have any tests. Ideally, I'd like all pull requests to have some tests.

File: usa-contacts/usagov.py (hunk 2 above)
Comment (jackiekazil) [inline usa-contacts/usagov.py]: I had a different setup earlier, so it has no use. I removed it... will check in an update shortly.

File: usa-contacts/usagov.py (hunk 1 above)
Comment (jackiekazil) [inline usa-contacts/usagov.py:55]: This has been updated.

File: usa-contacts/usagov.py (hunk 3 above)
Comment (jackiekazil) [inline usa-contacts/usagov.py:60]: While they are more brief and more computationally efficient, they are harder to quickly read.... so I am going to leave as is.

**Comment by jackiekazil** [comment]
@khandelwal While I agree, I have always handled testing in separate tickets -- we should discuss this as a team before it becomes a requirement to merge pull requests. If we make it an impromptu requirement for this ticket, then work on 18F/foia-hub#142 should be put on hold and the tests should be prioritized over that, because this merge effects work in 18F/foia-hub#142. I would argue we should hold off on major testing components until 142 is settled. @rjmajma?

**Comment by konklone** [comment]
I thought it was understood that tests would come in along with the code that they test. If that hasn't been understood, we can discuss it -- I'm comfortable saying ahead of time that it's a requirement I agree with. (And will not object to being scolded about, should I forget, as I certainly have.)

**Comment by khandelwal** [comment]
I'll let @rjmajma be the decider  - however I'd rather slow down and write _some_ tests.

**Comment by nsinai** [comment]
What would Mike Bland say?

**Comment by rjmajma** [comment]
I don't know what Mike Bland would say, but I like tests (unfair, I know, since I don't have to write them). There are two things being discussed: 
1. Whether to include tests within the pull request or as a separate ticket.
2. Whether this current pull request needs a ticket. 

Generally, I think it makes more sense to include the test as a part of the pull request itself. That being said, I don't have a horse in the race.

Since this PR doesn't integrate with anything, let's not prioritize writing a test for the time being and get to it if we can. Going forward, I would like for us to err on the side of including tests. This should be considered a built-in expectation of completion of an individual task.

**Comment by khandelwal** [comment]
We can start using https://github.com/cucumber/cucumber/wiki/Gherkin and then you can write all the tests @rjmajma

**Comment by cmc333333** [comment]
Tests would make this better, but let's not let perfect be the enemy of the good.

**Pull request merged**